packages/botonic-plugin-contentful/src/contentful/contents/follow-up.ts

```diff
@@ -77,18 +77,23 @@ export class FollowUpDelivery {
   }
 
   async commonFields(entry: Entry<CommonEntryFields>, context: cms.Context) {
-    const common = ContentfulEntryUtils.commonFieldsFromEntry(entry)
+    const followUp = await this.getFollowUp(entry, context)
+    return ContentfulEntryUtils.commonFieldsFromEntry(entry, followUp)
+  }
+
+  private async getFollowUp(
+    entry: Entry<CommonEntryFields>,
+    context: cms.Context
+  ): Promise<cms.FollowUp | undefined> {
     if (entry.fields.followup) {
       const followUp = entry.fields.followup.sys.contentType
         ? entry.fields.followup
         : await this.delivery.getEntry<FollowUpFields>(
             entry.fields.followup.sys.id,
             context
           )
-
-      common.followUp = await this.fromEntry(followUp, context)
-      return common
+      return (await this.fromEntry(followUp, context)) as cms.FollowUp
     }
-    return common
+    return undefined
```

Comment:
Why do we prefer to return `undefined` rather than `null`?

Reply:
I decided to use undefined everywhere rather than null at the beginning of the contentful project based on my ignorance on js. In the future I may decide to change the policy